code/modules/hazmat/crystal_processor.dm

```diff
@@ -0,0 +1,55 @@
+#define CRYSTAL_MACHINE_COLOR "colorizer"
+#define CRYSTAL_MACHINE_SHAPE "shaper"
+#define CRYSTAL_MACHINE_SIZE "sizer"
+/obj/machinery/hazmat/crystal_processor
+	name = "crystal processor"
+	desc = "Modify whatever crystal you got."
```

Comment:
should be more descriptive, something like "takes crystals and modifies their properties etc etc"